src/components/RewardWrapper/index.tsx

```diff
@@ -360,6 +363,18 @@ export function RewardWrapper({
       throw Error('no address found when trying to mint')
     }
 
+    const existingSignature = await getExistingSignature(
```

Comment:
is there a way we can move this out of the claim function? It will make the claim less responsive and the user won't see the existing signature dialog until after they click claim